libraries/chain/util/smt_token.cpp

```diff
@@ -23,17 +23,39 @@ const smt_token_object* find_smt_token( database& db, uint32_t nai )
    return nullptr;
 }
 
-const smt_token_object* find_smt_token( database& db, asset_symbol_type symbol, bool precision_agnostic )
+const smt_token_object* find_token( const database& db, asset_symbol_type symbol, bool precision_agnostic )
 {
    // Only liquid symbols are stored in the smt token index
    auto s = symbol.is_vesting() ? symbol.get_paired_symbol() : symbol;
 
    if ( precision_agnostic )
-      return find_smt_token( db, s.to_nai() );
+      return find_token( db, s.to_nai() );
    else
       return db.find< smt_token_object, by_symbol >( s );
 }
 
-} } } // steem::chain::util
+fc::optional< time_point_sec > last_emission_time( const database& db, const asset_symbol_type& symbol )
+{
+   const auto& idx = db.get_index< smt_token_emissions_index >().indices().get< by_symbol >();
```

Comment:
This can be shortened and made more readable by using the chainbase `get_index` templates.

`db.get_index< smt_token_emissions_index, by_symbol >()`